### PR TITLE
Developer validations and tests

### PR DESCRIPTION
**Pull request by gummibeatz**
Developer whitelist added.
Redirects to root if user is not on the dev whitelist
wrote developer and model tests

### CODE REVIEW

File: spec/models/developer_spec.rb [hunk 1]
```diff
@@ -3,11 +3,16 @@
 RSpec.describe Developer, type: :model do
 
   it { should have_many(:exit_tickets) }
-
-  it "creates a developer" do
```
Comment (mikekavouras) [inline spec/models/developer_spec.rb]: Why is this deleted? We still want to make sure that creating a developer works.

File: app/models/user.rb [hunk 2]
```diff
@@ -65,17 +66,20 @@ def self.from_google_omniauth(auth_hash)
...
-    developer = Developer.where(github_username: data["nickname"]).first
-    user = developer.user
-
-    unless user
-      user = developer.create_user(name: developer.full_name,
-         email: developer.email,
-         image: data["image"],
-         password: Devise.friendly_token[0,20]
-      )
+    user = User.create(email:"failed_email@email.com",
```
Comment (mikekavouras) [inline app/models/user.rb]: Is this supposed to be in here?

File: app/models/developer.rb [hunk 3]
```diff
@@ -20,6 +27,12 @@ def display_name
     full_name.split(" ").map(&:capitalize).join(" ")
   end
 
+  def whitelisted_github_username
+    if !DEVELOPER_WHITELIST.include?(github_username)
```
Comment (mikekavouras) [inline app/models/developer.rb]: this is a good time to use `unless`

File: app/models/developer.rb [hunk 4]
```diff
@@ -13,6 +17,9 @@ class Developer < ActiveRecord::Base
   belongs_to :cohort
 
   validates :email, uniqueness: true
+  validate :whitelisted_github_username
```
Comment (mikekavouras) [inline app/models/developer.rb]: This won't allow new developers to get created? What if there are no developers in the db. How would the first dev get validated???

Comment (gummibeatz) [inline app/models/developer.rb]: I thought we didn't want them to get created? Or we would have to append a new list to the developer whitelist

File: spec/models/developer_spec.rb (hunk 1 above)
Comment (gummibeatz) [inline spec/models/developer_spec.rb]: I grouped the creations with the validation a

Comment (gummibeatz) [inline spec/models/developer_spec.rb]: I'll change that as well though

File: app/models/developer.rb (hunk 3 above)
Comment (gummibeatz) [inline app/models/developer.rb]: I'll update it

File: app/models/user.rb (hunk 2 above)
Comment (gummibeatz) [inline app/models/user.rb]: I was trying to redirect to root with a user error when the developer didn't have access. 
The failed user would have the errors. I'll change it though
